defold-rive/editor/src/rive.clj

```diff
@@ -336,116 +354,230 @@
-  (let [handle (get-in renderable [:user-data :rive-file-handle])
-        vertices (rive-file->vertices handle 0.0)
-        mesh {:vertices vertices :indices nil}
-        mesh (transform-vertices (:world-transform renderable) mesh)]
-    mesh))
-
-(defn generate-vertex-buffer [renderables]
-  (let [meshes (map renderable->mesh renderables)
-        vcount (reduce + 0 (map (fn [mesh] (count (:vertices mesh))) meshes))]
-    (when (> vcount 0)
-      ; vertices are vec3-vec2-vec4
-      (let [vb (render/->vtx-pos-tex-col vcount)
-            verts (into [] (reduce concat (map (fn [mesh] (map vec (:vertices mesh))) meshes)))]
-        ; verts should be in the format [[x y x u v r g b a] [x y x u v r g b a] ...]
-        (persistent! (reduce conj! vb verts))))))
-
-; TODO: Load shader resources by name from the extension!
-
-(shader/defshader rive-id-vertex-shader
-  (attribute vec4 position)
-  (attribute vec2 texcoord0)
-  (varying vec2 var_texcoord0)
-  (defn void main []
-    (setq gl_Position (* gl_ModelViewProjectionMatrix position))
-    (setq var_texcoord0 texcoord0)))
 
-(shader/defshader rive-id-fragment-shader
-  (varying vec2 var_texcoord0)
-  ;(uniform sampler2D texture_sampler)
-  (uniform vec4 id)
-  (defn void main []
-    (setq gl_FragColor id)))
-    ;; (setq vec4 color (texture2D texture_sampler var_texcoord0.xy))
-    ;; (if (> color.a 0.05)
-    ;;   (setq gl_FragColor id)
-    ;;   (discard))
+(set! *warn-on-reflection* false)
 
+(defn- transform-vertices-as-vec [vertices]
+  ; vertices is a float array with 2-tuples (x y)
+  (map (fn [vert] [(.x vert) (.y vert)] ) vertices))
+
+(set! *warn-on-reflection* true)
 
-(def rive-id-shader (shader/make-shader ::id-shader rive-id-vertex-shader rive-id-fragment-shader {"id" :id}))
+(defn- renderable->handle [renderable]
+  (get-in renderable [:user-data :rive-file-handle]))
 
-(shader/defshader rive-shader-ver-tex-col
-  (attribute vec4 position)
-  (attribute vec2 texcoord0)
-  (attribute vec4 color)
-  (varying vec2 var_texcoord0)
-  (varying vec4 var_color)
+
+(shader/defshader rive-id-shader-vp
+  (attribute vec2 position)
+  (uniform vec4 cover)
   (defn void main []
-    (setq gl_Position (* gl_ModelViewProjectionMatrix position))
-    (setq var_texcoord0 texcoord0)
-    (setq var_color color)))
+    (setq vec4 pos (vec4 position.xy 0.0 1.0))
+    (setq gl_Position (+ (* (* gl_ModelViewProjectionMatrix pos) (- 1.0  cover.x)) (* cover.x pos)))
+    ))
 
-(shader/defshader rive-shader-frag-tint
-  (varying vec2 var_texcoord0)
-  (varying vec4 var_color)
+(shader/defshader rive-id-shader-fp
+  (uniform vec4 color)
+  (uniform vec4 id)
   (defn void main []
-    (setq gl_FragColor var_color)))
+    (if (> color.a 0.05)
+      (setq gl_FragColor id)
+      (discard))
+    ))
 
-(def rive-shader-tint (shader/make-shader ::shader rive-shader-ver-tex-col rive-shader-frag-tint))
+(def rive-id-shader (shader/make-shader ::id-shader rive-id-shader-vp rive-id-shader-fp {"id" :id}))
```

Comment:
This isn't actually working yet.
But we'll postpone fixes until we have proper mesh rendering.